### PR TITLE
Add TokenReadableKVState in hedera-mirror-web3

### LINKED ISSUE #9252: Implement `TokenDatabaseReadableKVState`
### Problem

We should add implementation for `TokenDatabaseReadableKVState`, so that we can return tokens in PBJ format from DB.

### Solution

Add `TokenDatabaseReadableKVState` in the following manner:

- Copy `TokenDatabaseAccessor` logic as a starting point
- Add repository logic for reading tokens from DB
- Add a new converter between entity and token DTO types to PBJ, so that the returned object is compliant with the hedera.app state

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by IvanKavaldzhiev**
**Description**:
Adding TokenReadableKVState for returning fungible token models into PBJ types, reading from DB.

**Related issue(s)**:

Fixes #9252 

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [X] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/state/TokenReadableKVStateTest.java [hunk 1]
```diff
@@ -0,0 +1,770 @@
...
+    }
+
+    @Test
+    void getTokenKeysValues() {
+        when(contractCallContext.getTimestamp()).thenReturn(Optional.empty());
+        setupToken(Optional.empty());
+        when(commonEntityAccessor.get(TOKEN_ID, Optional.empty())).thenReturn(Optional.ofNullable(entity));
+
+        assertThat(tokenReadableKVState.readFromDataSource(TOKEN_ID)).satisfies(token -> assertThat(token)
+                .returns(Utils.parseKey(entity.getKey(), entity.getId()), Token::adminKey)
```
Comment (edwin-greene) [inline hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/state/TokenReadableKVStateTest.java:344]: This does not compile for me locally. Does the branch need to be merged/updated upstream somewhere?

```
hedera-mirror-node/hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/state/TokenReadableKVStateTest.java:344: error: method parseKey in class Utils cannot be applied to given types;
                .returns(Utils.parseKey(entity.getKey(), entity.getId()), Token::adminKey)
                              ^
  required: byte[]
  found:    byte[],Long
  reason: actual and formal argument lists differ in length
  ```

File: hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/state/TokenReadableKVStateTest.java [hunk 2]
```diff
@@ -0,0 +1,770 @@
...
+        final var token = tokenReadableKVState.readFromDataSource(TOKEN_ID);
+        var results = token.customFees();
+        var customFee = results.get(0);
+        var resultFee = customFee.royaltyFee();
+        assertEquals(
+                royaltyFee.getNumerator(), resultFee.exchangeValueFraction().numerator());
+        assertEquals(
+                royaltyFee.getDenominator(), resultFee.exchangeValueFraction().denominator());
+        assertEquals(
+                0L, resultFee.fallbackFee() != null ? resultFee.fallbackFee().amount() : 0L);
```
Comment (edwin-greene) [inline hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/state/TokenReadableKVStateTest.java:518]: Could this be simplified to assert that fallbackFee is null?

File: hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/state/TokenReadableKVStateTest.java (hunk 1 above)
Comment (edwin-greene) [reply hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/state/TokenReadableKVStateTest.java:344]: I see, it needs 9414 merged first.

**Comment by kselveliev** [comment]
Need to merge main into this branch since AccountReadableKVState logic is already in main

**Comment by kselveliev** [comment]
Closing this in favor of https://github.com/hashgraph/hedera-mirror-node/pull/9502

**Pull request closed without merging**